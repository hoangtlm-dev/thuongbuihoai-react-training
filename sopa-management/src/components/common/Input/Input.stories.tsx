// Libs
import type { Meta, StoryObj } from '@storybook/react';

// Components
import Input,
{
  InputTheme,
  InputType
} from '.';

const meta: Meta<typeof Input> = {
  component: Input,
};

export default meta;
type Story = StoryObj<typeof Input>;

export const InputDefault: Story = {
  args: {
    label: 'Email address',
    type: InputType.default
  }
};

export const InputInfo: Story = {
  args: {
    label: 'Email address',
    type: InputType.info,
    theme: InputTheme.info,
  }
};

export const InputError: Story = {
  args: {
    label: 'Email address',
    type: InputType.error,
    theme: InputTheme.error,
  }
};
