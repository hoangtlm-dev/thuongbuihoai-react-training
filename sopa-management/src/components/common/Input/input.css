.input-wrapper {
  position: relative;
}

.input {
  border: 1px solid #02021D;
  border-radius: 16px;
  position: relative;
  width: 450px;
  padding: 20px 9px;
}

.input-info {
  border: 1px solid #3184FF;
}

.input-error {
  border: 1px solid #F14E2E;
}

.label {
  position: absolute;
  top: -8px;
  z-index: 1;
  left: 32px;
  background-color: white;
  padding: 0 5px;
  border: none;
}

.label-info {
  color: #3184FF;
}

.label-error {
  color: #F14E2E;
}