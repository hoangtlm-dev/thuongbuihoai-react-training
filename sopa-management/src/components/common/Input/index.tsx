import React, { memo } from 'react';

// Styles
import './input.css';

export enum InputType {
  default = '',
  info = 'info',
  error = 'error'
}

export enum InputTheme {
  default = '',
  info = 'info',
  error = 'error'
}

interface IconProps {
  label?: string;
  type?: string;
  theme?: string;
  placeholder?: string;
  classNameInput?: string;
  classNameLabel?: string;
}

const Input: React.FC<IconProps> = ({
  label = '',
  type = InputType.default,
  theme = InputTheme.default,
  placeholder = '',
  classNameInput = 'input',
  classNameLabel = 'label'
}) => (
  <div data-testId='input-value' className='input-wrapper'>
    <label className={`${classNameLabel} label-${theme}`}>{label}</label>
    <input
      placeholder={placeholder}
      className={`${classNameInput} input-${type}`}
    />
  </div>
);

export default memo(Input);
