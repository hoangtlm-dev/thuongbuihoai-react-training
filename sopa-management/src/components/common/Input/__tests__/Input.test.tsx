import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import Input from '../index';

describe('Input component', () => {
  test('Should render Input component', () => {
    const { getByTestId } = render(
      <Input type='text' placeholder='Enter email...' />
    );
    const inputValue = getByTestId('input');
    expect(inputValue).toBeInTheDocument();
  });

  test('Should render Input component by default', () => {
    const { getByTestId } = render(<Input />);
    const inputValue = getByTestId('input');
    expect(inputValue).toBeInTheDocument();
  });

  test('Should render Input component with label', () => {
    const { getByTestId } = render(<Input label='Products' />);
    const inputValue = getByTestId('input-value');
    expect(inputValue).toBeInTheDocument();
  });

  test('Check value input', async () => {
    render(<Input type='text' placeholder='Enter email...' />);
    const inputElement = screen.getByPlaceholderText(
      /Enter email.../i
    ) as HTMLInputElement;
    fireEvent.change(inputElement, {
      target: { value: 'ht@gmail.com' }
    });
    expect(inputElement.value).toBe('ht@gmail.com');
  });

  test('Matches snapshot', () => {
    const { container } = render(
      <Input type='text' placeholder='Your email...' />
    );
    expect(container).toMatchSnapshot();
  });
});
