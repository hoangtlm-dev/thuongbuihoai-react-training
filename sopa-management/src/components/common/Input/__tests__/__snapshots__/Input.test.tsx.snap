// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Matches snapshot 1`] = `
<div>
  <div
    class="input-wrapper"
    data-testid="input-value"
  >
    <label
      class="label label-"
    />
    <input
      class="input input-text"
      placeholder="Your email..."
    />
  </div>
</div>
`;
